# Project Euler.net

## Multiples of 3 and 5
   
### Problem 1

If we list all the natural numbers below 10 that are multiples of `3` or `5`, we get `3`, `5`, `6` and `9`. The sum of these multiples is `23`.

Find the sum of all the multiples of `3` or `5` below `1000`.


In [2]:
 def sum_mul_3_5(number):
        numbers = []
        for num in range(1, number):
            if (num%3 == 0) or (num%5 ==0):
                numbers.append(num)
        return sum(numbers)

In [3]:
print(sum_mul_3_5(10))
print(sum_mul_3_5(1000))

23
233168


#### Problem 1 Answer: `233168`
Completed on **Sat**, 25 Jan 2020, 17:02

## Even Fibonacci numbers
### Problem 2
Each new term in the Fibonacci sequence is generated by adding the previous two terms. By starting with `1` and `2`, the first `10` terms will be:

`1, 2, 3, 5, 8, 13, 21, 34, 55, 89, ...`

By considering the terms in the Fibonacci sequence whose values **do not exceed four million**, find the sum of the even-valued terms.


In [4]:
def fibonacci_sequence(number):
    terms = [1,2]
    total = 0
    for num in range(2, number):
        total = terms[num-2] + terms[num-1]
        terms.append(total)
    return terms
fibonacci_sequence(35)
    

[1,
 2,
 3,
 5,
 8,
 13,
 21,
 34,
 55,
 89,
 144,
 233,
 377,
 610,
 987,
 1597,
 2584,
 4181,
 6765,
 10946,
 17711,
 28657,
 46368,
 75025,
 121393,
 196418,
 317811,
 514229,
 832040,
 1346269,
 2178309,
 3524578,
 5702887,
 9227465,
 14930352]

In [5]:
total = []
for num in fibonacci_sequence(400):
    if num %2 == 0:
        if num < 4000000:
            total.append(num)
print(sum(total))

4613732


#### Problem 2 Answer: `233168`
Completed on **Sat**, 25 Jan 2020, 18:57

## Largest prime factor   
### Problem 3

The prime factors of `13195` are `5`, `7`, `13` and `29`.

What is the largest prime factor of the number `600851475143` ?


In [30]:
def find_prime(number):
    for num in range(2,number+1):
        if number%num != 0:
            return 'prime'
        else:
            return 'not prime'
    return 'prime'

In [31]:
def find_large_prime(number):
    primes = []
    for num in range(1, number):
        if find_prime(num) == 'prime':
            if number%num == 0:
                primes.append(num)
    return primes

NameError: name 'prime' is not defined